azurerm/resource_arm_application_gateway_test.go

```diff
@@ -0,0 +1,1265 @@
+    port = 8080
+  }
+
+  backend_address_pool {
+    # id = computed
+    name = "pool-1"
+
+    fqdn_list = [
+      "terraform.io",
+    ]
+  }
+
+  backend_http_settings {
+    # id = computed
+    name                  = "backend-http-1"
+    port                  = 8010
+    protocol              = "Https"
+    cookie_based_affinity = "Enabled"
+    request_timeout       = 30
+
+    # probe_id = computed
+    probe_name = "probe-1"
+  }
+
+  http_listener {
+    # id = computed
+    name = "listener-1"
+
+    # frontend_ip_configuration_id = computed
+    frontend_ip_configuration_name = "ip-config-public"
+
+    # frontend_ip_port_id = computed
+    frontend_port_name = "port-8080"
+    protocol           = "Http"
+  }
+
+  http_listener {
+    name                           = "listener-2"
+    frontend_ip_configuration_name = "ip-config-public"
+    frontend_port_name             = "port-8080"
+    protocol                       = "Https"
+
+    # ssl_certificate_id = computed
+    ssl_certificate_name = "ssl-1"
+    host_name            = "terraform.io"
+    require_sni          = true
+  }
+
+  probe {
+    # id = computed
+    name                = "probe-1"
+    protocol            = "Https"
+    path                = "/test"
+    host                = "azure.com"
+    timeout             = 120
+    interval            = 300
+    unhealthy_threshold = 8
+  }
+
+  url_path_map {
+    # id = computed
+    name                               = "path-map-1"
+    default_backend_address_pool_name  = "pool-1"
+    default_backend_http_settings_name = "backend-http-1"
+
+    path_rule {
+      # id = computed
+      name                       = "path-rule-1"
+      backend_address_pool_name  = "pool-1"
+      backend_http_settings_name = "backend-http-1"
+
+      paths = [
+        "/test",
+      ]
+    }
+  }
+
+  request_routing_rule {
+    # id = computed
+    name      = "rule-basic-1"
+    rule_type = "Basic"
+
+    # http_listener_id = computed
+    http_listener_name = "listener-1"
+
+    # backend_address_pool_id = computed
+    backend_address_pool_name = "pool-1"
+
+    # backend_http_settings_id = computed
+    backend_http_settings_name = "backend-http-1"
+  }
+
+  request_routing_rule {
+    # id = computed
+    name              = "rule-path-1"
+    rule_type         = "PathBasedRouting"
+    url_path_map_name = "path-map-1"
+
+    # http_listener_id = computed
+    http_listener_name = "listener-2"
+  }
+
+  ssl_certificate {
+    # id = computed
+    name     = "ssl-1"
+    data     = "${file("resource_arm_application_gateway_test.pfx")}"
+    password = "terraform"
+  }
+
+  tags {
+    environment = "tf01"
+  }
+}
+`, rInt, rInt)
+}
+
+func testAccAzureRMApplicationGateway_basic_changeSslCert(rInt int) string {
+	return fmt.Sprintf(`
+resource "azurerm_resource_group" "test" {
+  name     = "acctestrg-%d"
+  location = "West US 2"
+}
+
+resource "azurerm_virtual_network" "test" {
+  name                = "vnet"
```

Comment:
(same as above) can we make this a unique network name? (we tend to prefix them with `acctest` - which will become more important in the near future)